packages/react-core/src/components/Form/__tests__/FormGroup.test.tsx

```diff
@@ -203,4 +204,70 @@ describe('FormGroup', () => {
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('input should receive focus when clicking label', () => {
+    render(
+      <FormGroup data-testId="form-group-test-id" label="label" fieldId="native-label-element">
+        <input id="native-label-element" />
+      </FormGroup>
+    );
+
+    const labelElement = screen.getByTestId('form-group-test-id').querySelector('.pf-c-form__label');
+    const input = screen.getByTestId('form-group-test-id').querySelector('input');
+
+    userEvent.click(labelElement);
+    expect(input).toHaveFocus();
+  });
+
+  describe('With multiple inputs per group', () => {
+    test('inputs should not receive focus when clicking label with multipleInputs passed in', () => {
+      render(
+        <FormGroup label="label" fieldId="span-label-element" multipleInputs={{ isRadioGroup: false }}>
+          <input id="span-label-element" />
+          <input />
+        </FormGroup>
+      );
+
+      const labelElement = screen.getByRole('group').querySelector('.pf-c-form__label');
+      const inputs = screen.getByRole('group').querySelectorAll('input');
```

Comment:
And for selecting multiple inputs you should be able to just use `getAllByRole` rather than `getByRole`.